web/app/scripts/views/map/layers-directive.js

```diff
@@ -28,12 +28,69 @@
                                                {attribution: 'PRS'});
             map.addLayer(recordsLayer, {detectRetina: true});
 
+            // interactivity for record layer
+            var utfGridRecordsLayer = new L.UtfGrid(Config.windshaft.hostname +
+                                                    '/tiles/table/ashlar_record/id/ALL/{z}/{x}/{y}.grid.json',
+                                                    {useJsonP: false});
+            map.addLayer(utfGridRecordsLayer, {detectRetina: true});
+
+            utfGridRecordsLayer.on('click', function(e) {
+                var popupOptions = {
+                    maxWidth: 400,
+                    maxHeight: 300,
+                    autoPan: true,
+                    closeButton: true,
+                    autoPanPadding: [5, 5]
+                };
+
+                var popup = new L.popup(popupOptions)
+                    .setLatLng(e.latlng)
+                    .setContent(buildRecordPopup(e.data))
+                    .openOn(map);
+            });
+
             // user-uploaded boundary layer(s)
             var boundaryLayer = new L.tileLayer(Config.windshaft.hostname +
                                                 '/tiles/table/ashlar_boundary/id/ALL/{z}/{x}/{y}.png',
                                                 {attribution: 'PRS'});
             map.addLayer(boundaryLayer, {detectRetina: true});
         }
+
+        /**
+         * Build popup content from arbitrary record data.
+         *
+         * @param {Object} UTFGrid interactivity data from interaction event object
+         * @returns {String} HTML snippet for a Leaflet popup.
+         */
+        function buildRecordPopup(record) {
```

Comment:
There's not really a way to test this directly that I know of, as it is a private function, but I'll see if I can find a way to test it indirectly by inspecting the generated element.